src/modules/schedule/model.ts

```diff
@@ -0,0 +1,28 @@
+import mongoose, { Schema, InferSchemaType, Types } from "mongoose";
+
+export const CustomEventSchema = new Schema({
+  start_time: { type: String, required: true },
+  end_time: { type: String, required: true },
+  title: { type: String, required: false },
+  location: { type: String, required: false },
+  description: { type: String, required: false },
+  days_of_week: { type: String, required: false }
+});
+
+
+export const ScheduleSchema = new Schema({
+  _id: { type: Types.ObjectId, reqired: true },
+  name: { type: String, required: false },
+  created_by: { type: String, required: true },
+  term: { type: String, required: true },
+  public: { type: Boolean, required: true },
+  class_IDs: { type: [String], required: false },
+  primary_section_IDs: { type: [String], required: false },
+  secondary_section_IDs: { type: [String], required: false },
+  custom_events: { type: [CustomEventSchema], required: false }
```

Comment:
Remove _id. 

Might be good to consider defaults and other [options](https://mongoosejs.com/docs/schematypes.html#schematype-options).